src/constructLayoutEngine.js

```diff
@@ -159,10 +159,11 @@ function recurseRoutes({
       if (shouldMount) {
         if (!route.connectedNode) {
           const newNode = route.cloneNode(false);
-          insertNode(newNode, parentContainer, previousSibling);
           route.connectedNode = newNode;
         }
 
+        insertNode(route.connectedNode, parentContainer, previousSibling);
```

Comment:
This fixes a bug in the layout engine where dom elements wouldn't be rearranged during route transitions if they were mounted both before and after.